viamonvif/gosoap/gosoap.go

```diff
@@ -0,0 +1,229 @@
+// Package gosoap provides a minimal soap client for viamonvif
+package gosoap
+
+import (
+	//nolint: gosec
+	"crypto/sha1"
+	"encoding/base64"
+	"encoding/xml"
+	"time"
+
+	// TODO: Remove dependence on etree.
+	"github.com/beevik/etree"
+	"github.com/elgs/gostrgen"
+)
+
+// SoapMessage type from string.
+type SoapMessage string
+
+// NewEmptySOAP return new SoapMessage.
+func NewEmptySOAP() (SoapMessage, error) {
+	var zero SoapMessage
+	doc := etree.NewDocument()
+	doc.CreateProcInst("xml", `version="1.0" encoding="UTF-8"`)
+	env := doc.CreateElement("soap-env:Envelope")
+	env.CreateElement("soap-env:Header")
+	env.CreateElement("soap-env:Body")
+	env.CreateAttr("xmlns:soap-env", "http://www.w3.org/2003/05/soap-envelope")
+	env.CreateAttr("xmlns:soap-enc", "http://www.w3.org/2003/05/soap-encoding")
+	res, err := doc.WriteToString()
```

Comment:
We should migrate this in the future to just use XML struct annotations given the vast majority of the XML document will be the same and we only need to make 4 XML requests...